### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by iw-gtadam**
* Move out cloudwatch and alert manager ports to secrets manager
* Restructure TF files
* Use common template for ECS task definitions
* Update documentation diagram with cloudwatch exporter

### CODE REVIEW

File: alertmanager_s3.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+data template_file "alertmanager" {
+  count    = local.is_management_env ? 1 : 0
+  template = file("${path.module}/config/alertmanager/config.yml")
+  vars = {
+    slack_api_url = jsondecode(data.aws_secretsmanager_secret_version.dataworks[local.primary_role_index].secret_binary)["slack_webhook_url"]
+    http_proxy    = "http://${aws_vpc_endpoint.internet_proxy[0].dns_entry[0].dns_name}:3128"
```
Comment (ChrisScottThomas) [inline alertmanager_s3.tf:7]: Should this port also go into secrets?

File: alertmanager_sg.tf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "allow_alertmanager_egress_https" {
+  count             = local.is_management_env ? 1 : 0
+  description       = "Allows ECS to pull container from S3"
+  type              = "egress"
+  to_port           = 443
```
Comment (ChrisScottThomas) [inline alertmanager_sg.tf:17]: Another port?

File: cloudwatch_exporter_ecs.tf [hunk 3]
```diff
@@ -0,0 +1,67 @@
+resource "aws_ecs_task_definition" "cloudwatch_exporter" {
+  family                   = "cloudwatch-exporter"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = "512"
+  memory                   = "4096"
```
Comment (ChrisScottThomas) [inline cloudwatch_exporter_ecs.tf:6]: aren't these set in `variables.tf`?

File: cloudwatch_exporter_ecs.tf [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      }
+    ])
+  }
+}
+
+resource "aws_ecs_service" "cloudwatch_exporter" {
+  name             = "cloudwatch-exporter"
+  cluster          = local.is_management_env ? data.terraform_remote_state.management.outputs.ecs_cluster_main.id : data.terraform_remote_state.common.outputs.ecs_cluster_main.id
+  task_definition  = aws_ecs_task_definition.cloudwatch_exporter.arn
+  platform_version = "1.4.0"
```
Comment (ChrisScottThomas) [inline cloudwatch_exporter_ecs.tf:41]: If we parameterise these, does that keep more people in line with versioning?

File: cloudwatch_exporter_ecs.tf (hunk 3 above)
Comment (iw-gtadam) [reply cloudwatch_exporter_ecs.tf:6]: These are the limits for the task not the amount used for the the amount used in the container. Unfortunately the can sometimes have to be increased based on the the number of sidecars.

File: alertmanager_s3.tf (hunk 1 above)
Comment (iw-gtadam) [reply alertmanager_s3.tf:7]: Yeah will amend this

Comment (iw-gtadam) [reply alertmanager_s3.tf:7]: resolved

File: alertmanager_sg.tf (hunk 2 above)
Comment (iw-gtadam) [reply alertmanager_sg.tf]: Resolved

File: cloudwatch_exporter_ecs.tf (hunk 4 above)
Comment (iw-gtadam) [reply cloudwatch_exporter_ecs.tf]: I believe this is resolved

**Pull request merged**